zcash_client_sqlite/src/wallet.rs

```diff
@@ -920,36 +1160,20 @@ fn subtree_scan_progress<P: consensus::Parameters>(
             .optional()?
             .flatten()
             .map(Ok)
-            .or_else(|| {
-                stmt_start_tree_size
-                    .query_row(
-                        named_params![":start_height": u32::from(birthday_height)],
-                        |row| row.get::<_, Option<u64>>(0),
-                    )
-                    .optional()
-                    .map(|opt| opt.flatten())
-                    .transpose()
-            })
+            // If we don't have an explicit birthday tree size, find something nearby.
+            .or_else(|| get_tree_size_near(birthday_height).transpose())
             .transpose()?;
 
-        // Get the note commitment tree size as of the end of the recover-until height.
+        // Get the note commitment tree size as of the start of the recover-until height.
         let recover_until_size = recover_until_height
```

Comment:
This is an `Option<Option<u64>>`. Are both levels of `Option` meaningful or can they be flattened?

Reply:
Both are meaningful:
- The outer optional means "do we have a recover-until height?", and is `unwrap_or`ed to the corresponding birthday height values as relevant.
- The inner optional means "have we scanned enough information from the chain to get tree size data for the recover-until height?", and is used for triggering fallback behaviour.

Reply:
Please add a comment about this.